src/test/java/org/jitsi/meet/test/base/AbstractBaseTest.java

```diff
@@ -107,6 +112,8 @@ protected AbstractBaseTest(AbstractBaseTest baseTest)
         participants = new ParticipantHelper(baseTest.participants);
     }
 
+    public abstract ParticipantFactory getFactory(Properties config);
```

Comment:
Previous code was selecting factory based on the participant type. This way a test was able to create both mobile and web participants. That's not possible anymore.

Reply:
Ok, I see what you mean. If we separate stuff and we have web and mobile extending base, I think it's good to not import web or mobile in base. I will look for solution of this.

Reply:
I don't see why this would be a problem here. It's not Jitsi OSGi bundle nor we're looking to have any plugin system here, but I'm curious what will you come up with.